source/extensions/filters/common/expr/context.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "envoy/stream_info/stream_info.h"
+
+#include "common/http/headers.h"
+
+#include "eval/public/cel_value.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+
+using CelValue = google::api::expr::runtime::CelValue;
+
+// Symbols for traversing the request properties in the expressions
+constexpr absl::string_view Request = "request";
+constexpr absl::string_view Path = "path";
+constexpr absl::string_view Host = "host";
+constexpr absl::string_view Scheme = "scheme";
+constexpr absl::string_view Method = "method";
+constexpr absl::string_view Referer = "referer";
+constexpr absl::string_view Headers = "headers";
+constexpr absl::string_view Time = "time";
+constexpr absl::string_view ID = "id";
+constexpr absl::string_view UserAgent = "useragent";
+constexpr absl::string_view Size = "size";
+constexpr absl::string_view TotalSize = "total_size";
+
+// Per-request metadata
+constexpr absl::string_view Metadata = "metadata";
+
+// Downstream connection properties
+constexpr absl::string_view Connection = "connection";
+constexpr absl::string_view LocalAddress = "local_address";
+constexpr absl::string_view RemoteAddress = "remote_address";
+
```

Comment:
So all this is restricted to just HTTP? I could use the existing sql parser with rbac to do stuff based on queries. If you look at the sql filter, it emits the query type [insert/update/select/delete] and the db.table name.

Reply:
You can access any dynamic metadata, using `metadata.filter_metadata["filter namespace"]`. I don't see where sql stuff is reflected in stream info. If it is, let me know and I can populate it. The tricky part is string ownership, since the interpreter does not own anything except what is on the arena.